be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -318,14 +330,44 @@ Status Segment::_load_index_impl() {
     });
 }
 
+static vectorized::DataTypePtr get_data_type_from_column_meta(
+        const segment_v2::ColumnMetaPB& column) {
+    return vectorized::DataTypeFactory::instance().create_data_type(column);
+}
+
+vectorized::DataTypePtr Segment::get_data_type_of(const Field& field, bool ignore_children) const {
+    // Path has higher priority
+    if (!field.path().empty()) {
+        auto node = _sub_column_tree.find_leaf(field.path());
+        if (node) {
+            if (ignore_children || node->children.empty()) {
+                return node->data.file_column_type;
+            }
+        }
+        // it contains children or column missing in storage, so treat it as variant
+        return field.is_nullable()
+                       ? vectorized::make_nullable(std::make_shared<vectorized::DataTypeObject>())
+                       : std::make_shared<vectorized::DataTypeObject>();
+    }
+    // TODO support normal column type
+    return nullptr;
```

Comment:
It's not intuitive to return nullprt of normal column. May be we can change the function name related to variant.

Reply:
it's an interface for both variant and none variant column, in the future, implement normal column for `light schema change data type`

Reply:
return `nullptr` meaning it's no need to be compared with storage type

Reply:
add comment 'no such column'

Reply:
done